pkg/controller/utils/pod/pod.go

```diff
@@ -101,12 +101,13 @@ func GetPodConditionFromList(conditions []v1.PodCondition, conditionType v1.PodC
 
 // IsPodRestarting checks if a pod in the Canary deployment is restarting
 // This returns the "reason" for the pod with the most restarts
-func IsPodRestarting(pod *v1.Pod) (bool, datadoghqv1alpha1.ExtendedDaemonSetStatusReason) {
-	var maxRestartCount int
+func IsPodRestarting(pod *v1.Pod, maxRestartCount int) (bool, datadoghqv1alpha1.ExtendedDaemonSetStatusReason) {
+	// track the highest number of restarts among pod containers
+	var mostRestartCount int
```

Comment:
what if the eds controller restarts during a canary? I think we should store the count in the CR status

Reply:
`mostRestartCount` is calculated by looking at each run  the `pod.Status`, so I think we are safe if the controller restart: the function is completely stateless.